stactools_landsat/stactools/landsat/utils.py

```diff
@@ -0,0 +1,124 @@
+import datetime
+
+import dateutil
+import rasterio
+from pystac import Item, Link
+from rasterio import RasterioIOError
+from shapely.geometry import box, mapping, shape
+
+
+def _parse_date(in_date: str) -> datetime.datetime:
```

Comment:
Use [pystac.utils.datetime_to_str](https://github.com/stac-utils/pystac/blob/develop/pystac/utils.py#L144) (has more generality and is better tested)

Reply:
I see. I'll look into it, but I think I see what you're saying. Parse the date here using the generic tool, and make it a string using the `datetime_to_str` function.

Reply:
Ack, nope. Sorry, got my wires crossed, through this was going datetime -> str, it's going str -> datetime.

[pystac.utils.str_to_datetime](https://github.com/stac-utils/pystac/blob/develop/pystac/utils.py#L164-L165) just uses the dateutil parse, and this is setting the TZ, so disregard my comment!